libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1079,10 +1081,24 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	{
 		// No initial value is only permitted for single variables with specified type.
 		if (_statement.declarations().size() != 1 || !_statement.declarations().front())
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			// Bailing out *fatal* here, as those (untyped) vars may be used later, and diagnostics wouldn't be helpful then.
+			if (!boost::algorithm::all_of_equal(_statement.declarations(), nullptr))
+				m_errorReporter.fatalTypeError(_statement.location(), "Use of the \"var\" keyword is disallowed.");
+			else
+			{
+				// It is okay to simply return here, as there are no named components on the
```

Comment:
I'd move the comments around. The `okay to return` should go before the `return` statement and the assertion should have the explanation that "ensure that an error was emitted already. TODO: actually check the right one was emitted"

Reply:
@axic wrt. your TODO-statement, I'd propose to do that in a seperate PR because it would need an extension of the `ErrorReporter` class. Do you agree?

Reply:
I'm not sure whether it's even necessary to check which kind of error was raised before. What would be the benefit?

Reply:
That the correct error was raised for this case and we are not just checking that *any* error exists.